src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -159,8 +159,15 @@ trait RuntimeCosting extends CostingRules with DataCosting with Slicing { IR: IR
   }
 
   def costOf(costOp: CostOf, doEval: Boolean): Rep[Int] = {
-    val res = if (doEval) toRep(costOp.eval)
-    else (costOp: Rep[Int])
+    val res = if (doEval) {
+      val c = Const(costOp.eval)
+      // optimized hot-spot: here we avoid rewriting which is done by reifyObject
+      findOrCreateDefinition(c, c.self)
```

Comment:
Please add function comments where it is implemented

Reply:
Done